File: Logging/LogEntry.cs
﻿using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;

namespace ova.Common.Logging
{
    public class LogEntry
    {
        public LogEntry()
        {
            long _timestamp = UnixTime.Base.Timestamp(); //DateTime.UtcNow;
            UserName = Environment.UserName;
        }

        static public readonly string StaticHostName = System.Net.Dns.GetHostName(); ///  сделать через HOST

        public string UserName { get; private set; }
        public string HostName { get { return StaticHostName; } }
        public long TimeStampUnix { get; private set; }
        public string Category { get; set; }
        public LogLevel Level { get; set; }
        public string Text { get; set; }
        public Exception Exception { get; set; }
        public EventId EventId { get; set; }
        public object State { get; set; }
        public string StateText { get; set; }
        public Dictionary<string, object> StateProperties { get; set; }
        public List<LogScopeInfo> Scopes { get; set; }
    }

    public class LogScopeInfo       // LogScopeInfo represents Scope information regarding a LogEntry
    {
        public LogScopeInfo() { }

        public string Text { get; set; }
        public Dictionary<string, object> Properties { get; set; }
    }

}

File: DbLogging/LogEntry.vb
Imports Microsoft.Extensions.Logging


Friend Class LogEntry
    Public Sub New()
        TimeStamp = TimeStamp()
        UserName = Environment.UserName
    End Sub

    Public Shared ReadOnly StaticHostName As String = Net.Dns.GetHostName()
    Public ReadOnly Property HostName As String
    Public ReadOnly Property UserName As String
    Public Property TimeStamp As Long
    Public Property Category As String
    Public Property Level As LogLevel
    Public Property Message As String
    Public Property Exception As Exception
    Public Property EventId As LogEvents
    Public Property State As Object
    Public Property StateText As String
    Public Property StateProperties As Dictionary(Of String, Object)
    Public Property Scopes As List(Of LogScopeInfo)

End Class

Friend Class LogScopeInfo
    Public Property Text As String
    Public Property Properties As Dictionary(Of String, Object)
End Class

